### ISSUE TITLE
Multivariate Restructure

### ISSUE THREAD

**Issue by farzanmrz** [farzanmrz/spread-res#38]
Restructure for multiple metadata changing fundamental way of a lot of things listed:

1. **TorchDS Dataloader:** How the data is represented in our custom Dataset object cause all metadata comes in now
2. **Model:** Internally the forward method splits into 2 methods; hidden to get till basically the projected hidden state encoding and then forward runs from main training where we attach individual heads for each metadata, So essentially the hidden I think remains same and in forward we add differing layers
3. **Training:** THe function changes to CELoss, architectural changes where loss for each individual head is taken together (unsure if add or concat) and then backpropogated through the model so it starts becoming a generalist over various different metadata.

Way more things here just confused for now so leaving this as is
